UnitySDK/Assets/ML-Agents/Scripts/Sensor/VectorSensor.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace MLAgents.Sensor
+{
+    public class VectorSensor : SensorBase
+    {
+        public List<float> observations;
```

Comment:
why not `float[]`?  This would do out of bounds checking and raise exceptions automatically.

Reply:
Is there any performance benefit? I like the current approach of appending the observations and clearning after writing.

Reply:
less memory fragmentation.

Reply:
for small observation spaces it's not a huge deal, for something like crawler, or walker, it could be pretty bad.